mlearning/tflite-micro/tflm_syslog.cc

```diff
@@ -0,0 +1,66 @@
+/****************************************************************************
+ * apps/mlearning/tflite-micro/tflm_syslog.cc
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#include "tensorflow/lite/micro/micro_log.h"
+
+#include <syslog.h>
+#include <stdio.h>
+
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
+#include "tensorflow/lite/micro/debug_log.h"
+#endif
+
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
+
+void VMicroPrintf(FAR const char *format, va_list ap)
+{
+  vsyslog(CONFIG_TFLITEMICRO_SYSLOG_LEVEL, format, ap);
+}
+
+void MicroPrintf(FAR const char *format, ...)
```

Comment:
why not implement DebugLog like other porting?

Reply:
tfmico provide the standard method to overwrite the log channel, so it's better to follow their design.

Reply:
OK, maybe we can add a NuttX backend

Reply:
@xiaoxiang781216 tflm common code adds CRLF processing. If we redirect micro log to the debug log, unnecessary blank lines will appear in the print:

```
[    4.570000] [ 4] 0 (id=0): size=16, offset=0, first_used=0 last_used=1
[    4.570000] [ 4] 
[    4.570000] [ 4] 1 (id=1): size=64, offset=64, first_used=1 last_used=2
[    4.570000] [ 4] 
[    4.570000] [ 4] 2 (id=2): size=64, offset=0, first_used=2 last_used=3
[    4.570000] [ 4] 
[    4.570000] [ 4] 3 (id=3): size=16, offset=64, first_used=3 last_used=3
[    4.570000] [ 4] 
[    4.570000] [ 4]  0: 0000000000...................................................................... (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  1: 0000000000..............................1111111111111111111111111111111111111111 (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  2: 22222222222222222222222222222222222222221111111111111111111111111111111111111111 (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  3: 22222222222222222222222222222222222222223333333333.............................. (1k)
```

Maybe the current implementation is a better choice?

Reply:
@xiaoxiang781216

Reply:
ok.